typewise-alert.c

```diff
@@ -11,61 +12,49 @@ BreachType inferBreach(double value, double lowerLimit, double upperLimit) {
   return NORMAL;
 }
 
-BreachType classifyTemperatureBreach(
-    CoolingType coolingType, double temperatureInC) {
-  int lowerLimit = 0;
-  int upperLimit = 0;
-  switch(coolingType) {
-    case PASSIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 35;
-      break;
-    case HI_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 45;
-      break;
-    case MED_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 40;
-      break;
-  }
-  return inferBreach(temperatureInC, lowerLimit, upperLimit);
-}
+void checkAndAlert(BatteryCharacter batteryChar, double temperatureInC)
+{
 
-void checkAndAlert(
-    AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC) {
-
-  BreachType breachType = classifyTemperatureBreach(
-    batteryChar.coolingType, temperatureInC
-  );
-
-  switch(alertTarget) {
-    case TO_CONTROLLER:
-      sendToController(breachType);
-      break;
-    case TO_EMAIL:
-      sendToEmail(breachType);
-      break;
-  }
+  BreachType breachType = inferBreach(temperatureInC,
+                                      batteryChar.Threshold.lowerLimit,
+                                      batteryChar.Threshold.upperLimit);
+        batteryChar.Alert(breachType);
 }
 
 void sendToController(BreachType breachType) {
   const unsigned short header = 0xfeed;
   printf("%x : %x\n", header, breachType);
 }
 
+char *BreachTypeStr[BreachType_TotalNumber] = {
+  "too low",
+  "too high",
+  "normal"
+};
+
+char *BreachTypeToString(BreachType breachType)
+{
+  assert((breachType < BreachType_TotalNumber)&&(breachType >= NORMAL));
+  return BreachTypeStr[breachType];
+}
+
 void sendToEmail(BreachType breachType) {
   const char* recepient = "a.b@c.com";
   switch(breachType) {
-    case TOO_LOW:
+    default:
```

Comment:
This is a roundabout way of saying 'if not normal, print the message'
Why do we need a switch-case for it?

Reply:
/* say if warning thresholds are introduced for which i dont need to print anymsg
    then i can simply add a case statement below normal like
      >>> case NORMAL:
      >>> case LOWERWARNINGTRHESHOLD: 
      >>>  break;
    so error msg doesnt gets printed and it doesnt get clumsier and easier to read
    */
    
    i have entered my view in the comments .. i think i have not tried my best to mention my intention.
    
    The code was already with the switch case and i thought about the above mentioned usecase.
    i could say i had used the domain knowledge of TCQ ...lol..
    Normally you give us an example and you later extend it up (BMS ) was a very good example.
So i thought you might add some more thresholds to this and then i should be using  not normal and not lower warning threshold and not higher warning threshold .. and it becomes ugly. so i followed my laziness and stopped changing it.


But early in the morning i come to remember a short discussion which we had
"Martin fowler states that never do speculative coding" - gonna follow that.